en/guides/typesafehttpbasic.md

```diff
@@ -0,0 +1,172 @@
+```swift
+public struct MyBasicAuth: TypeSafeHTTPBasic {
+
+}
+```
+2) Add protocol stubs:
+
+Xcode should appear with the message:
+`Type 'MyBasicAuth' does not conform to protocol 'TypeSafeCredentials'`
+
+Click fix to autogenerate the stubs and produce the below struct:
+```swift
+public struct MyBasicAuth: TypeSafeHTTPBasic {
+    public static var verifyPassword: ((String, String, @escaping (MyBasicAuth?) -> Void) -> Void)
+
+    public var id: String
+}
+```
+
+3) Inside MyBasicAuth, add an authentication dictionary:
+
+
+```swift
+public static let authenticate = ["username" : "password"]
+```
+
+__Note:__ In a real project, never store passwords in plain text!
+
+4) Define verifyPassword:
+
+The closure, `verifyPassword`, takes a username and password and returns a `MyBasicAuth` instance on success. We want to check if the password matches the user's stored password. On successful match, we initialise `MyBasicAuth` with an `id` equal to username.
+
+```swift
+public static var verifyPassword: ((String, String, @escaping (MyBasicAuth?) -> Void) -> Void) =
+   { userId, password, callback in
+       if let storedPassword = authenticate[userId], storedPassword == password {
+           callback(MyBasicAuth(id: userId))
+       } else {
+           callback(nil)
+       }
+   }
+```
+
+5) Your complete struct should now look as follows:
+```swift
+public struct MyBasicAuth: TypeSafeHTTPBasic {
+
+    public static let authenticate = ["username" : "password"]
+
+    public static var verifyPassword: ((String, String, @escaping (MyBasicAuth?) -> Void) -> Void) =
+    { userId, password, callback in
+        if let storedPassword = authenticate[userId], storedPassword == password {
+            callback(MyBasicAuth(id: userId))
+        } else {
+            callback(nil)
+        }
+    }
+    public var id: String
+}
+```
+
+## Using your MyBasicAuth in a route
+
+1) Inside postInit(), add the following route:
+```swift
+router.get("/basic") { (user: MyBasicAuth, respondWith: (MyBasicAuth?, RequestError?) -> Void) in
+    print("authenticated \(user.id) using \(user.provider)")
+    respondWith(user, nil)
+}
+```
+
+2) Go to [http://localhost:8080/basic](http://localhost:8080/basic)
+3) Login with User Name: username, Password: password
+4) You will successfully login and be returned your username
+
+Congratulations, you have just implemented HTTP basic authentication of a Codable route!!!
+
+The browser will store your login credentials and automatically log you in if you return to the route. Use a private window if you would like to test failed authentication.
+
+## Using TypeSafeHTTPBasic with the ORM
+
+This example is a simple struct just containing the required id. A real world project would probably have a user profile with lots of fields which is keyed by the unique id. We will show you how to implement this is using [Swift-Kuery-ORM](https://github.com/IBM-Swift/Swift-Kuery-ORM).
```

Comment:
-> "This example shows a simple struct that contains only the required id."

We will show you how to implement this is using [Swift-Kuery-ORM](https://github.com/IBM-Swift/Swift-Kuery-ORM). -> "We demonstrate how to implement this using [Swift-Kuery-ORM](https://github.com/IBM-Swift/Swift-Kuery-ORM)."